gard_owl_ingest/analysis/mondo_mapping_status.py

```diff
@@ -1,7 +1,11 @@
-"""Mapping status between GARD and Mondo"""
+"""Mapping status between GARD and Mondo
```

Comment:
@matentzn @sabrinatoro @nicolevasilevsky Here are the results of my latest analysis regarding alignment between GARD and Mondo:
- [Spreadsheet](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=244617092)
- [Readme](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit) (It's not very good but I did write down some things I discovered)

The gist of it is that last week I took a look at the new GARD csv they supplied us, and I tried to discover which obsolete GARD terms still exist in Mondo. That list is [here](https://docs.google.com/spreadsheets/d/1dfaG3Ry-15mhJ3_nwi30MvC_kqJNJUk44wlbLxV72dw/edit#gid=1163401515), though GARD is still reviewing, so I don't think we're yet removing these mappings from Mondo.

Then, this week, I tried to see if we could obtain new GARD::Mondo mappings using Orphanet as a proxy. That is to say, GARD has almost 100% of its terms mapped to Orphanet. And we also have a whole lot of Orphanet mappings. So, using that common Orphanet term, we are ascertaining new mappings. Previously, we had about 1,600 GARD terms in Mondo. Now, through this proxy method, the mappings are at about ~7,000 (of a total 12,000) GARD terms.

Reply:
This looks great. Can you export a legal sssom mapping for GaRD-ORDO and GARD-OMIM as part of this process?

Also if "almost all" GARD codes have an ORDO mapping - can we say that _all_ GARD codes are mapped to either a omim or Ordo? 

If true, I would have expected 99% coverage for Mondo, because the progress monitor say https://monarch-initiative.github.io/mondo-ingest/reports/unmapped/

We are missing less than 2%?

Reply:
### GARD::ORDO
For GARD::ORDO, I think we're assuming `skosExactMatch` for everything.

### GARD::OMIM

### GARD::OMIM
For GARD::OMIM, however, it's not so clear. GARD terms have 0 to many OMIM mappings. For the ones that have 2+ mappings, I do not know what kind of match they are. I could consider them unspecified for now. But I wonder what you @matentzn think/suggest. If it helps, you can look at examples some of the current mappings in the [spreadsheet](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=244617092).

### All GARD terms mapped?
> Also if "almost all" GARD codes have an ORDO mapping - can we say that all GARD codes are mapped to either a omim or Ordo?

Doesn't look like it. There is a "DataSource" column with 3 possible values: "Orphanet", "Orphanet+OMIM", and "GARD"
<img width="1214" alt="Screen Shot 2023-04-14 at 12 22 38 PM" src="https://user-images.githubusercontent.com/13045020/232101085-ef86c0b5-9f19-4615-b480-442475e448b7.png">
I just now took a look at all the cases where `"DataSource" == "GARD"`, and I'm surprised to see there are some things in the `OmimMember` column. I don't know why the DataSource is not "OMIM" in that case.

But you can see that there are some that do not have mappings.

Reply:
Can you update your numbers:

Given a complete Mondo-ORDO and Mondo-OMIM mapping, what is the % of GARD terms we can faithfully map to Mondo using the two above?

Reply:
By complete, I assume you mean that if we hypothetically had 100% of Orphanet and OMIM terms mapped to Mondo, how well could we map to GARD?

The GARD CSV they provided has 12,410 terms. Out of those, 12,400 are mapped to ORDO. The remaining 10 are in the screenshot above. Of those 10, 5 have OMIM mappings, and 5 do not. So, that would mean we could map 12,405/12,410 terms, or 99.96%. Pretty good!

Reply:
If Mondo has all of Orphanet (which it supposedly does see your progress tracker) and GARD is nearly 100% mapped to Orphanet we should be too. Can you create a sample of 50 Orphanet terms GARD is mapped with, but we are not? And share it with Sabrina for checking.

Reply:
I was mistaken. See Sabrina's issue I responded to / email I sent to GARD. The SourceIDs in the Orphanet+OMIM subset appear to be MIM#s, not Orphanet IDs as I'd suspected.

Reply:
Even then, since we are mapped 100% with OMiM and 100% with Orphanet we should be at 100% with GARD, no?

Reply:
I concur it should be much higher. I just double checked [my comment](https://github.com/monarch-initiative/mondo-ingest/issues/283#issuecomment-1512316246) in Sabrina's issue. Even with this change in the analysis I'll be doing, it looks like there's still about 2,400 terms unaccounted for. I won't know the reason for that until I dig in further.

Reply:
@matentzn @sabrinatoro New results:

([spreadsheet](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=1607652739) | [google doc](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit))
Unmapped GARD terms: 193 of 12004 (1.61%)
Mapped GARD terms: 11811 of 12004 (98.39%)

---

I fixed the issue where I was accidentally mapping to ORDO IDs that should have been OMIM IDs. I also found some new columns to map to. I versioned the [spreadsheet](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=1607652739), then updated it. The main sheet we want to look at now is [gard.sssom](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=1504531712).

I think you guys should review at my notes in the [readme](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit) to see how I determined mappings. I also added comments to some column headers explaining its contents and how I interpreted those fields.

I imagine you guys will be happy by this much higher mapping %! But also you will wonder why the number of unmapped GARD terms (n=193) is higher than the unmapped ORDO+OMIM terms (n=130). I copied the table below from [the progress monitor](https://github.com/monarch-initiative/mondo-ingest/blob/main/docs/reports/unmapped.md).

| Ontology | Tot unmapped (mappable) |
|:----------|:-----|
| ORDO       | 124  |
| OMIM       | 6      |

Of the 12,410 GARD terms, 5 of them terms have no Orphanet or OMIM mapping. But the unmapped number is still higher than expected. Some reasons why this could be happening:
a. Perhaps GARD is mapping to some deprecated terms.
b. Perhaps GARD is mapping to some terms not showing up in our ingest for some reason.
c. Probably some other reason, but I'm not sure yet.

I think probably at this stage, figuring the rest of these mappings out is probably better done by manual curation than scripting, but if there is more work I can do that you guys think would be helpful, or if you notice errors in any of my [assumptions](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit#), I can fix the script and rerun.

Reply:
@matentzn @sabrinatoro New results:

([spreadsheet](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=1607652739) | [google doc](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit))
Unmapped GARD terms: 193 of 12004 (1.61%)
Mapped GARD terms: 11811 of 12004 (98.39%)

---

I fixed the issue where I was accidentally mapping to ORDO IDs that should have been OMIM IDs. I also found some new columns to map to. I versioned the [spreadsheet](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=1607652739), then updated it. The main sheet we want to look at now is [gard.sssom](https://docs.google.com/spreadsheets/d/1w5Xnzr5uNFcPrQqCT8mGBFHnGhwXDfQVzHKofw6kB7c/edit#gid=1504531712).

I think you guys should review at my notes in the [readme](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit) to see how I determined mappings. I also added comments to some column headers explaining its contents and how I interpreted those fields.

I imagine you guys will be happy by this much higher mapping %! But also you will wonder why the number of unmapped GARD terms (n=193) is higher than the unmapped ORDO+OMIM terms (n=130). I copied the table below from [the progress monitor](https://github.com/monarch-initiative/mondo-ingest/blob/main/docs/reports/unmapped.md).

| Ontology | Tot unmapped (mappable) |
|:----------|:-----|
| ORDO       | 124  |
| OMIM       | 6      |

Of the 12,410 GARD terms, 5 of them terms have no Orphanet or OMIM mapping. But the unmapped number is still higher than expected. Some reasons why this could be happening:
a. Perhaps GARD is mapping to some deprecated terms.
b. Perhaps GARD is mapping to some terms not showing up in our ingest for some reason.
c. Probably some other reason, but I'm not sure yet.

I think probably at this stage, figuring the rest of these mappings out is probably better done by manual curation than scripting, but if there is more work I can do that you guys think would be helpful, or if you notice errors in any of my [assumptions](https://docs.google.com/document/d/1sM0B0ICdOWDwZWt0M_Y3t5QYwqL-XwOsDl3TS14XRa4/edit#), I can fix the script and rerun.